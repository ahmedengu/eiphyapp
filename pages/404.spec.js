import { mount } from 'enzyme';
import { axe } from 'jest-axe';
import Custom404 from './404';

jest.mock('../components/header', () => () => <header>Header</header>);

function setup(props = {}) {
  return mount(<Custom404 msg="this is an error" {...props} />);
}

describe('Custom404 Tests', () => {
  it('should work', () => {
    const actual = setup();
    expect(actual).toBe(actual);
  });

  it('should match snapshot', async () => {
    const actual = setup();
    const html = actual.html();
    expect(html).toMatchSnapshot();
    expect(await axe(html)).toHaveNoViolations();
  });
});
