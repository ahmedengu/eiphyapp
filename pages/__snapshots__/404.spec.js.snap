// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom404 Tests should match snapshot 1`] = `
<div id="divLoading"
     class="show"
>
</div>
<div id="layout">
  <header>
    Header
  </header>
  <main class="main-layout">
    <h1>
      404 - Page Not Found
    </h1>
  </main>
  <footer class="mastfoot mt-auto">
    <div class="inner">
    </div>
  </footer>
</div>
`;
